server/src/main/java/org/candlepin/dto/manifest/v1/ProductDTO.java

```diff
@@ -908,7 +1011,9 @@ public int hashCode() {
             .append(this.getAttributes())
             .append(this.getDependentProductIds())
             .append(this.getBranding())
-            .append(pcHashCode);
+            .append(this.getProvidedProducts())
+            .append(pcHashCode)
+            .append(ppHashCode);
```

Comment:
Same here, no need to calculate hashcode above and set it here. This line is enough: `.append(this.getProvidedProducts())`.